src/app/components/core/configuration-manager/list-type-configuration/list-type-configuration.component.ts

```diff
@@ -69,8 +102,48 @@ export class ListTypeConfigurationComponent implements OnInit {
           return num;
         });
       }
-      this.form.get(this.configuration.key)?.patchValue(JSON.stringify(value))
-      this.formState.emit(this.listItems.valid);
+      if(this.configuration.items == 'object') {
+        let listValues = this.generateListValuesArray(value);
+        this.changedConfig.emit({ [this.configuration.key]: JSON.stringify(listValues) });
+      }
+      else{
+        this.changedConfig.emit({ [this.configuration.key]: JSON.stringify(value) });
+      }
+      this.formStatusEvent.emit({'status': this.listItems.valid, 'group': this.group});
+      // this.form.get(this.configuration.key)?.patchValue(JSON.stringify(value))
+      // this.formState.emit(this.listItems.valid);
     })
   }
+
+  getChangedConfiguration(index: string, propertyChangedValues: any) {
```

Comment:
seems it will always be the last index of the list items?

Reply:
No, here index is the index of list items array. This is required to change the corresponding list item.